### ISSUE TITLE
CodeBuild  to deploy to EKS - giving error "You must be logged in to the server (Unauthorized)"

### ISSUE THREAD

**Issue by sibendu** [kubernetes-sigs/aws-iam-authenticator#329]
Hi All,
I am trying to deploy to EKS from CodeBuild. It is using a service role "codebuild-checking-service-role". I have tried adding this service role in configmap aws-auth. I have tried adding it under mapRoles: section, as well as also tried with it uner mapUsers: section (following some sugestions). But in both case it is giving an error - You must be logged in to the server (Unauthorized)

Since it is creating a dynamic role binding , is it sufficient to add on service role arn i.e. for the caller identity it shows arn:aws:sts::11111111111:assumed-role/codebuild-checking-service-role/AWSCodeBuild-b8e28caf-28f5-4eec-bda5-a3ea4174da26 ; since it is dynamic value how will it work?

 
2020/07/30 18:16:18 Running command **aws sts get-caller-identity** --
20 | {
21 | "UserId": "AROA2TWYKNW7S2GT3EEBP:AWSCodeBuild-b8e28caf-28f5-4eec-bda5-a3ea4174da26",
22 | "Account": "11111111111",
23 | "Arn": "arn:aws:sts::11111111111:assumed-role/codebuild-checking-service-role/AWSCodeBuild-b8e28caf-28f5-4eec-bda5-a3ea4174da26"
24 | }
25 |  
26 | [Container] 2020/07/30 18:16:24 Running command **aws eks update-kubeconfig --name <---> --region <--->**
27 | Added new context arn:aws:eks:ap-south-1:729524366783:cluster/eks-digital to /root/.kube/config
28 |  
29 | [Container] 2020/07/30 18:16:24 Running command **aws eks update-kubeconfig --name <---> --region <---> --role-arn arn:aws:iam::11111111111:role/service-role/codebuild-checking-service-role**
30 | Updated context arn:aws:eks:ap-south-1:729524366783:cluster/eks-digital in /root/.kube/config
31 |  
32 | [Container] 2020/07/30 18:16:25 Running command **kubectl get nodes**
33 | error: You must be logged in to the server (Unauthorized)

Thanks a lot for any suggestion. 
Regards.

**Comment by ArtoKeskiniva** [comment]
I had the very same issue - and finally found a resolution here: https://github.com/kubernetes-sigs/aws-iam-authenticator/issues/268
The point is to be **remove extra path component** from role ARN in configmap.

**Comment by sibendu** [comment]
Absolutely precise. Thank you very much @ArtoKeskiniva

**Comment by sibendu** [comment]
resolution : [#268](https://github.com/kubernetes-sigs/aws-iam-authenticator/issues/268)

**Comment by barrychum** [comment]
Hi,
I ran into a similar situation and received the same error.  I am also trying to deploy EKS from CodeBuild and `kubectl get svc` generated the same error `error: You must be logged in to the server (Unauthorized)`

I verified configmap that arn does not have the extra path.  Any help is appreciated.

CodeBuild build log
`32  [Container] 2021/06/11 09:15:36 Running command aws sts get-caller-identity`
`33	{`
`34	    "UserId": "AROA4MSYOBEVG6B2UFNJN:AWSCodeBuild-657071bb-01e0-478a-acca-bde5e88972c0",`
`35	    "Account": "111122223333",`
`36	    "Arn": "arn:aws:sts::111122223333:assumed-role/codebuild-eks/AWSCodeBuild-657071bb-01e0-478a-acca-bde5e88972c0"`
`37	}`
`38`	
`39	[Container] 2021/06/11 09:15:37 Running command aws eks update-kubeconfig --region eu-central-1 --name clusterdemo --role-arn arn:aws:iam::111122223333:role/codebuild-eks`
`40	Added new context arn:aws:eks:eu-central-1:111122223333:cluster/clusterdemo to /root/.kube/config`
`41	`
`42	[Container] 2021/06/11 09:15:38 Running command kubectl get nodes`
`43	error: You must be logged in to the server (Unauthorized)`

kubectl describe configmap -n kube-system aws-auth
`mapRoles:`
`----`
`- groups:`
`  - system:bootstrappers`
`  - system:nodes`
`  rolearn: arn:aws:iam::111122223333:role/eksctl-clusterdemo-nodegroup-ng1-NodeInstanceRole-7XQ5QA08SP1G`
`  username: system:node:{{EC2PrivateDNSName}}`
`- groups:`
`  - system:masters`
`  rolearn: arn:aws:iam::111122223333:role/codebuild-eks`
`  username:codebuild-eks`
